src/cli.py

```diff
@@ -19,25 +19,75 @@
 
 def cli():
+
+    sysVals = groundStation.SystemValues()
+
     while True:
         if flag.exit():
-            print('Exiting receiving loop')
+            print('Exiting receiving loop\n')
             flag.reset()
             return
         try:
             server, port, toSend = gs.getInput(prompt='to send: ')
-            if server == 24:
-                # This is a direct UART command to a ground station EnduroSat transceiver to enter PIPE
-                # Can be deleted for flight
-                gs.__setPIPE__()
+            data = bytearray(libcsp.packet_get_data(toSend))
+            cspData = bytearray(libcsp.packet_content(toSend))
+            print("toSend csp packet bytearray: ", cspData)
+
+            if (
+                server == sysVals.APP_DICT.get('OBC') and 
+                port == sysVals.SERVICES.get('SCHEDULER').get('port') and
+                data[0] == sysVals.SERVICES.get('SCHEDULER').get('subservice').get('SET_SCHEDULE').get('subPort')
+                ):
+
+                print('entered SET_SCHEDULE\n')
+
+                # open the scheduler text file as an array of strings
+                with open('schedule.txt') as f:
+                    cmdList = f.readlines()
+
+                #create an empty list, and create another list of csp objects
+                schedule = list()
+
+                # for each line of command, parse the packet
+                if len(cmdList) > 0:
+                    for i in range(0, len(cmdList)):
+
+                        # parse the time and command, then embed the command as a CSP packet
+                        schedulerObj = groundStation.embedCSP(cmdList[i])
+                        scheduler = schedulerObj.embedCSP()
+                        
+                        # append the list
+                        schedule.append(scheduler)
+                        print("list of schedules: ", schedule)
+                        scheduledTime = schedule[i]['time']
+                        scheduledDst = schedule[i]['dst']
+                        dst = (scheduledDst).to_bytes(2, byteorder='big')
+                        scheduledDport = schedule[i]['dport']
+                        dport = (scheduledDport).to_bytes(2, byteorder='big')
+                        scheduleSubservice = schedule[i]['subservice']
+                        returnPacket = bytearray(libcsp.packet_content(scheduleSubservice))
+                        data.extend(scheduledTime)
+                        data.extend(dst)
+                        data.extend(dport)
+                        data.extend(returnPacket)
+                    #reply = libcsp.buffer_get(len(scheduledTime)+1)
+                    libcsp.packet_set_data(toSend, data)
+                    #libcsp.sendto_reply(toSend, reply, libcsp.CSP_O_NONE)
+                    print("the server is: ", server)
+                    print("the port is: ", port)
+                    resp = gs.transaction(server, port, toSend)
```

Comment:
Not sure if it would be too much of a bother but can the embedded CSP packet creation relating to the scheduler be made as a class method of [groundStation](https://github.com/AlbertaSat/ex2_ground_station_software/blob/c2035f81a906161f351acb858c090f6f437e8f36/src/groundStation/groundStation.py#L134)? It would help avoid code repeat as the ground station web app relies on groundStation class methods (`getInput` and `transaction`) to execute commands, instead of the cli.

Reply:
absolutely